### ISSUE TITLE
bug: <title>

### ISSUE THREAD

**Issue by GeneralUser-NonDev** [ReVanced/revanced-patcher#244]
### Type

Other

### Bug description

I know this is most likely the wrong place to mention this but hopefully you can pass along to the correct destination.

the app (not sure if it's Revanced or MicroG) doesn't (and hasn't in FOREVER) kept track of watched videos in the "History" section, correctly. It picks and chooses what it wants to recall. I had the app shut down while watching a random video, I had to then try and recall the title to search for it again to finish it because it wasn't in the history.

### Steps to reproduce

1. watch a video
2. look at history section (under library tab)
3. see that the video you just watched is most likely not in the history list.

### Relevant log output

```shell
revanced on my mobile, not on my pc where i'm submitting this.
```

### Solution

fix/collaberate to fix the lack of video history recording 

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by GeneralUser-NonDev** [comment]
> Are you logged into YouTube? It won't keep a watch history if you don't use an account.

yes, i'm logged-in in revanced. I think it's actually through MicroG because when I had to uninstall revanced to update it with a new YT apk I don't recall having to re-login.

**Comment by GeneralUser-NonDev** [comment]
> I cannot reproduce on my end. Videos always show up correctly in the watch history. If you are force closing the app while watching then that might be the cause.

I've never force closed it. It's done it ever since I started using the original Vanced. Are you rooted, I've gone the non-root route, I don't know if that matters.

**Comment by GeneralUser-NonDev** [comment]
Nope, Watch History is on.

Doing a little more testing I have Adguard on both my Mobile and an Adguard Home server on my home network, While connected to the server (Home WiFi) it appears to not keep track of my watch history (In fact Adguard on my mobile shows that it blocked ads in Revanced, I Thought Revanced was supposed to do that itself). when I disabled my home WiFi (used cellular data) it appears to keep tabs of the videos again with or without adguard protection on my mobile. Any way to figure how how it works so I can see about putting in exceptions or something?

**Comment by GeneralUser-NonDev** [comment]
adguard on my phone is it's own app that basically intercepts traffic to remove the ads and such, at least that's how I understand it, I don't know dev type stuff.

The adguard server has it's dns set for (I think it's Cloudflare?) 1.1.1.1, 1.0.0.1, and Google 8.8.8.8, 8.8.4.4. to my knowledge I believe they are both "unfiltered" DNS resolvers so my adguard server is the only thing doing ad filtering.

**Comment by GeneralUser-NonDev** [comment]
![adguard dns](https://github.com/ReVanced/revanced-patcher/assets/96662916/0082ed05-8bad-463f-a190-9b38bfb41653)
![Adguard Blocklist sample](https://github.com/ReVanced/revanced-patcher/assets/96662916/5e7f6fc3-5b01-4f9a-8eaf-07a2c2d0fe98)

again, I don't know specifics, but I believe the server does similar to the app by which all the traffic gets pointed to it (all my network has it's dns pointed to the adguard server, ex: 192.168.1.22) and it filters out the ads and such on it's way out to the internet via cloudflare and google in my case.

**Comment by GeneralUser-NonDev** [comment]
but it's fine with the app on cellular, its the server on my WiFi (screenshots) that limits/breaks it when connected. i tried looking to see what the app uses but couldn't find it.

**Comment by Sonico98** [comment]
Try whitelisting s.youtube.com

**Comment by GeneralUser-NonDev** [comment]
> Try whitelisting s.youtube.com

tried both that and s2, looking at my phones logs while disconnected from wifi it's blocking the S's but still working correctly.
